### PR TITLE
Display usd

### PR DESCRIPTION
**Pull request by cxy2015ca**
- product detail page, shop all page, cart items, checkout items will display the # and currency of the selected currency

TODO:
- ~~need to change total price for checkout~~ (Done by Xin)
- ~~implement localstorage clearing on new code version~~ (Done by Yu Tong)

Yu Tong edit:
- Completed localstorage clearing on new code version
- Added geolocation for default currency, tested using TunnelBear

image
image
image
image

### CODE REVIEW

**Review by yutongliuytl — APPROVED** [review]
Let's get it

**Review by xinruili07 — APPROVED** [review]

File: src/components/cadShippingMethod.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
+/* eslint-disable no-nested-ternary */
+import React from 'react';
```
Comment (cxy2015ca) [inline src/components/cadShippingMethod.js:2]: nit: is it absolutely necessary to have 2 components for shipping LOL maybe we can just make the price a dynamic component next time, this is fine for now

File: src/components/checkout.js [hunk 2]
```diff
@@ -221,8 +175,11 @@ const Payment = () => {
     checkoutItems.forEach((item) => {
       const desc = `${item.size}/${item.name}/${item.prodMetadata.colour}/${item.prodMetadata.item}`;
       skusList.push({
-        amount: item.price * 100,
-        currency: 'cad',
+        amount:
+          (state.currency === 'USD'
+            ? convertCadToUsd(item.price)
```
Comment (cxy2015ca) [inline src/components/checkout.js:180]: no need to convert here, item.priceUSD has been added to context

File: src/context/Provider.js [hunk 3]
```diff
@@ -4,12 +4,26 @@ import PropTypes from 'prop-types';
 import initialState from './initialState';
 import globalReducer from './reducer';
 
+const getCountry = async () => {
+  return fetch('http://ip-api.com/json')
```
Comment (cxy2015ca) [inline src/context/Provider.js:8]: if its not too much work maybe leveraging this browser built in geolocation is better style? idk
https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition
we can consider for next time

File: src/context/Provider.js [hunk 4]
```diff
@@ -4,12 +4,26 @@ import PropTypes from 'prop-types';
...
+    });
+};
+
 let localState;
 
 if (typeof window !== `undefined`) {
   localState = JSON.parse(localStorage.getItem('cart-items'));
-  if (localState) {
-    localState.visitedPage = [];
+  if (localState && localState.version !== '1.0.0') {
```
Comment (cxy2015ca) [inline src/context/Provider.js:25]: we should abstract the version into an environment variable, so then we update the .env file moving forward and don't have to keep track of this in the code, i feel its more standard that way
ie. localState.version !== ENV.version

Comment (yutongliuytl) [reply src/context/Provider.js:25]: yes that is a good point, will add the change!

File: src/context/Provider.js (hunk 3 above)
Comment (yutongliuytl) [reply src/context/Provider.js:8]: Right, I look into it and it seems like it gives the latitude and longitude so we might also need to find a way to know the countries boundaries, will consider it for next time:')

**Pull request merged**